File: Solving/18784/18784.cpp
// Problem No.: 18784
// Solver:      Jinmin Goh
// Date:        20200714
// URL: https://www.acmicpc.net/problem/18784

#include <iostream>
#include <cstdlib>
#include <list>
#include <array>
#include <atomic>
#include <algorithm>
#include <deque>
#include <iostream>
#include <iterator>
#include <map>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <vector>
#include <tuple>

using namespace std;

int main(){
    // this code makes input & output fast as scanf / printf
    // cin.tie(NULL);
    // ios_base::sync_with_stdio(false);
    // check following blog: https://starrykss.tistory.com/750

    vector<pair<int, int>> coordX, coordY;
    int n;
    scanf("%d", &n);
    for(int i = 0; i < n; i++){
        int x, y;
        scanf("%d %d", &x, &y);
        coordX.push_back({ x, y });
        // same result with: coordX.emplace_back(x, y);
        // pair<int, int> = temp(x, y);
        // coordX.push_back(temp);
        coordY.push_back({ y, x });
    }
    sort(coordX.begin(), coordX.end());
    sort(coordY.begin(), coordY.end());
    
    return 0;
}

File: Solving/18784/18784.py
# Problem No.: 18784
# Solver:      Jinmin Goh
# Date:        20200714
# URL: https://www.acmicpc.net/problem/18784

import sys

def main():
    n = int(input())
    coordinates = {}
    coordX = {}
    coordY = {}
    for _ in range(n):
        x, y = map(int, sys.stdin.readline().split())
        if x not in coordX:
            coordX[x] = [(x, y)]
        else:
            coordX[x].append((x, y))
        if y not in coordY:
            coordY[y] = [(x, y)]
        else:
            coordY[y].append((x, y))
        coordinates[(x, y)] = 0
    
    validX = []
    validY = []
    validDot = []

    for i in coordX:
        if len(coordX[i]) > 1:
            validX.append(i)
            for j in coordX[i]:
                coordinates[j] += 1
    for i in coordY:
        if len(coordY[i]) > 1:
            validY.append(i)
            for j in coordY[i]:
                coordinates[j] += 1
                if coordinates[j] == 2:
                    validDot.append(j)
    
    ans = 0
    
    if not validDot:
        print(0)
        return
    
    for dot in validDot:
        [x, y] = dot
        #print(x, y)
        xSum = 0
        ySum = 0
        for dotX in coordX[x]:
            xSum += abs(dot[1] - dotX[1])
        for dotY in coordY[y]:
            ySum += abs(dot[0] - dotY[0])
        ans += abs(xSum * ySum)
        
    print(ans)
    return

if __name__ == "__main__":
    main()